### ISSUE TITLE
Single Selection Mode

### ISSUE THREAD

**Issue by fifonik** [alexandernst/angular-multi-select#71]
I'm trying to replace normal select with the directive in single selection mode in a way similar to normal angular way. I have input items as an array (ctrl.input) and model as an object (ctrl.data) with one prop (ctrl.data.browser_id) that I'd like to bind to the directive.

```
  <div
    angular-multi-select="" 
    input-model="ctrl.input"
    button-label="<[ name ]>"
    item-label="<[ name ]>"
    tick-property="checked"

    selection-mode="single"
    output-model="ctrl.output"
    single-output-model="ctrl.data.browser_id"
    single-output-prop="id"
    preselect-prop="id"
    preselect-value="{{ctrl.data.browser_id}}"
  ></div>
```

And I faced a few problems:
1. When I remove "output-model" I got an error "$compile/nonassign?p0=undefined&p1=angularMultiSelect". Based on documentation this should be allowed as I use @single-output-model and @single-output-prop: "Instead of output-model, you can use single-output-prop and single-output-model attributes to assign the first item of the output data to a model/variable"
2. When the directive initialized the model prop is set to empty value (and nothing is checked by default). If I understand correctly it should honour it as @preselect-prop and @preselect-value are specified.

Just wonder what am I missing?
Thanks.

P.S.
http://plnkr.co/edit/doD2cDe08PpAsS3zPSm0?p=preview
I added @delay-start here so you can see that ctrl.data.browser_id value is changed from 2 to "" on initialization

**Comment by fifonik** [comment]
After posting the issue I've checked the source code and realised that preselect-value processed not 100% correct. So now preselected value should be an array, not string as it mentioned in documentation. 
So the following changes in my code fixes the second issue:

```
- preselect-value="{{ctrl.data.browser_id}}"
+ preselect-value="[{{ctrl.data.browser_id}}]"
```

However, probably you'd like to change preselect-value processing code to something like this:

```
            $scope.fillShadowModel = function() {
                $scope._shadowModel = angular.copy($scope.inputModel);
                $scope._enforceProps($scope._shadowModel);

                try {
-                   $scope.preselectValue = JSON.parse($scope.preselectValue);
+                   var preselectValue = JSON.parse($scope.preselectValue);
+                   $scope.preselectValue = (angular.isArray(preselectValue)) ? preselectValue : [preselectValue];
                } catch(e) {
-                   $scope.preselectValue = [$scope.preselectValue];
+                   $scope.preselectValue = '';
                }
```

**Comment by alexandernst** [comment]
Indeed, nice catch! Can you make a PR?

**Comment by alexandernst** [comment]
Closed by #73
